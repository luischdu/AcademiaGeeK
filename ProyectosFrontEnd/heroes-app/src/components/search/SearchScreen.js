import React, { useMemo } from "react";
import queryString from "query-string";
import HeroCard from "../Heroes/HeroeCard";
import { useLocation } from "react-router-dom";
import { getHeroesByName } from "../../selectors/getHeroesByName";
import { useForm } from "../../hook/useForm";

const SearchScreen = ({history}) => {
  const location = useLocation();

  const { q = "" } = queryString.parse(location.search);

  const heroesFiltered = useMemo(() => getHeroesByName(q), [q]);

  const [formValues, handleInputChange] = useForm({
    searchText: q,
  });
  const { searchText } = formValues;

  const handleSearch = (e) => {
    e.preventDefault();
    history.push(`?q=${searchText}`);
  };

  return (
    <div>
      <h1>Search Screen</h1>
      <hr />

      <div className="row">
        <div className="col-5">
          <h4> Search Form </h4>
          <hr />

          <form onSubmit={handleSearch}>
            <input
              type="text"
              placeholder="Find your hero"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={searchText}
              onChange={handleInputChange}
            />

            <button
              type="submit"
              className="btn m-1 btn-block btn-outline-primary"
            >
              Search...
            </button>
          </form>
        </div>

        <div className="col-7">
          <h4> Results </h4>
          <hr />

          {q === "" && <div className="alert alert-info">Search a hero</div>}

          {q !== "" && heroesFiltered.length === 0 && (
            <div className="alert alert-danger">
              There is no a hero with {q}
            </div>
          )}

          {heroesFiltered.map((hero) => (
            <HeroCard key={hero.id} {...hero} />
          ))}
        </div>
      </div>
    </div>
  );
};

export default SearchScreen;
