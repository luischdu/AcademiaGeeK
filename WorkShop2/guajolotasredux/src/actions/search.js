import { types } from "../types/types";

export const search = (search) => {
  return {
    type: types.searchProducto,
    payload: {
      search,
    },
  };
};

export const keepSearch = (product) => {
  return (dispatch) => {
    dispatch(search(product));
  };
};
