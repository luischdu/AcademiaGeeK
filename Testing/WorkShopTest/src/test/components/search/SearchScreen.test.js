import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';
import { SearchScreen } from '../../../components/search/SearchScreen';


describe('Pruebas en SearchScreen', () => {

    test('Mostrar valor inicial', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.alert-info').text().trim() ).toBe('Search a hero');
 
    })


    test('Debe coincidir el valor buscador y la solicitud', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={[`/search?q=flash`]}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        );

        expect( wrapper.find('input').prop('value') ).toBe(`flash`);
        expect( wrapper ).toMatchSnapshot();

    })
})