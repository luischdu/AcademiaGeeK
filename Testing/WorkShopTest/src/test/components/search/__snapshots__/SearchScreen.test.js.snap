// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchScreen Debe coincidir el valor buscador y el 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="flash"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="card ms-3 animate__animated animate__fadeIn"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          />
          <div
            className="col-md-8"
          >
            <img
              alt="Flash"
              className="card-img"
              src="./assets/heroes/dc-flash.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Flash
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Jay Garrick
                 
              </p>
              <p
                className="card-text-characters"
              >
                 
                Jay Garrick, Barry Allen, Wally West, Bart Allen
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                   
                  Flash Comics #1
                   
                </small>
              </p>
              <a
                href="/hero/dc-flash"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en SearchScreen Debe coincidir el valor buscador y la solicitud 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="flash"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="card ms-3 animate__animated animate__fadeIn"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          />
          <div
            className="col-md-8"
          >
            <img
              alt="Flash"
              className="card-img"
              src="./assets/heroes/dc-flash.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Flash
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Jay Garrick
                 
              </p>
              <p
                className="card-text-characters"
              >
                 
                Jay Garrick, Barry Allen, Wally West, Bart Allen
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                   
                  Flash Comics #1
                   
                </small>
              </p>
              <a
                href="/hero/dc-flash"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en SearchScreen Mostrar valor inicial 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en SearchScreen debe de mostrar a Batman y el input con el valor del queryString 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="card ms-3 animate__animated animate__fadeIn"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          />
          <div
            className="col-md-8"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Batman
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Bruce Wayne
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                   
                  Detective Comics #27
                   
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
